src/client_side_reply.cc

```diff
@@ -119,7 +119,7 @@ clientReplyContext::setReplyToError(
 #if USE_AUTH
     errstate->auth_user_request = auth_user_request;
 #endif
-    setReplyToError(method, errstate);
+    setReplyToError(failedrequest ? failedrequest->method : HttpRequestMethod(Http::METHOD_NONE), errstate);
 }
 
 void clientReplyContext::setReplyToError(const HttpRequestMethod& method, ErrorState *errstate)
```

Comment:
> If we proceed with more changes maybe it is better to make them for master branch not for bag15g

This is a very good point! Let's postpone this particular discussion until we start working on master. I will log my concerns here so that they are not as easily lost, but ignore them for now:


> Christos: The other clientReplyContext::setReplyToError method can be simplified to and the method parameter can retrieved from errstate->request object. Moreover in **all cases I checked** the errstate->request is the same with http->request object. So probably the HttpRequest object parameter can be removed from clientReplyContext::setReplyToError method

I hesitate saying "let's simplify" when you are not sure that the two objects are the same in all cases. Perhaps there is a technical way to prove that the two requests are the same beyond a reasonable doubt? If not, can we at least claim that errstate->request and http->request must be the same (i.e. that it is a reasonable invariant even if some buggy code violates it)?

Finally, setReplyToError(method...) assumes that http->request may not exist. Is that assumption false? Does errstate->request always exist?

Reply:
> ...but ignore them for now:

OK.
Should we post a patch to the customer? Something else?

Reply:
There is another [change request](https://github.com/measurement-factory/squid/pull/130#discussion_r689697916) to be resolved first.